src/components/Gradebook/index.jsx

```diff
@@ -435,11 +444,46 @@ export default class Gradebook extends React.Component {
                   closeText="Cancel"
                   body={(
                     <div>
-                      <h3>{this.state.modalModel[0].assignmentName}</h3>
-                      <Table
-                        columns={[{ label: 'Username', key: 'username' }, { label: 'Current grade', key: 'currentGrade' }, { label: 'Adjusted grade', key: 'adjustedGrade' }]}
-                        data={this.state.modalModel}
+                      <div>
+                        <div className="grade-history-header grade-history-assignment">Assignment: </div> <div>{this.state.modalAssignmentName}</div>
+                        <div className="grade-history-header grade-history-student">Student: </div> <div>{this.state.updateUserName}</div>
+                        <div className="grade-history-header grade-history-original-grade">Original Grade: </div> <div>{this.state.originalGrade}</div>
+                        <div className="grade-history-header grade-history-current-grade">Current Grade: </div> <div>{this.props.gradeOverrideCurrentPossibleGradedOverride}</div>
+                      </div>
+                      <StatusAlert
+                        alertType="danger"
+                        dialog="Error retrieving grade override history."
+                        open={this.props.errorFetchingGradeOverrideHistory}
+                        dismissible={false}
                       />
+                      { !this.props.errorFetchingGradeOverrideHistory && (
+                      <Table
+                        columns={GRADE_OVERRIDE_HISTORY_COLUMNS}
+                        data={[...this.props.gradeOverrides, {
+                          date: this.state.todaysDate,
+                          grader: this.state.updateUserName,
+                          reason: (<input
+                            type="text"
+                            name="reasonForChange"
+                            value={this.state.reasonForChange}
+                            onChange={value => this.onChange(value)}
```

Comment:
One teachable moment point (I promise I'll space these out better in the future). If you define the function you're calling like this
```javascript
  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
```
then this can become
```suggestion
                            onChange={this.onChange}
```
and you don't have to create a new function just to call another function with `this` set to the correct thing. This isn't particularly important, but understanding why this works can help you understand the difference between a `function` function and an `=>`/"arrow" function.